core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -158,6 +158,37 @@ private[spark] class HostLocalDirManager(
   }
 }
 
+private[spark] trait MigratableRDDResolver {
+
+  def getMigratableRDDBlocks(): Seq[ReplicateBlock]
+
+  def migrateBlock(block: ReplicateBlock): Boolean
+}
+
+private[spark] class BlockManagerPeerProvider(
```

Comment:
I like this a little more than the trait because the implementation is right here. But I'm not seeing needing a class for just this (also the implementation changed which is _ok_ but I want to make sure we're doing it intentionally).